extensions/import/src/controllers/mainController.ts

```diff
@@ -25,15 +25,23 @@ export default class MainController extends ControllerBase {
 	public deactivate(): void {
 	}
 
-	public activate(): Promise<boolean> {
-		const outputChannel = vscode.window.createOutputChannel(constants.serviceName);
-		new ServiceClient(outputChannel).startService(this._context);
-
-		managerInstance.onRegisteredApi<FlatFileProvider>(ApiType.FlatFileProvider)(provider => {
-			this.initializeFlatFileProvider(provider);
+	public async activate(): Promise<boolean> {
+		return new Promise<boolean>(async (resolve) => {
+			try {
+				/*
+				Registering the command before downloading the file to avoid the command not found error.
+				*/
+				managerInstance.onRegisteredApi<FlatFileProvider>(ApiType.FlatFileProvider)(provider => {
```

Comment:
I would probably move this under the await new ServiceClient, weird behavior could occur otherwise

Reply:
I think you want to keep this here otherwise we might miss the onRegisteredApi callback since the ServiceClient is what kicks off the API being registered.